In [1]:
# Imports
import pandas as pd
import numpy as np
import re

In [15]:
def trace_filter_abf(array):
    pattern = r"\d{4}_\d{2}_\d{2}_\d{4}\.abf"
    filtered = [item for item in array if re.match(pattern+'.abf', str(item))]
    return np.array(filtered)

### Names of each annotation sheet

In [2]:
sheet_names_df = pd.read_csv('/Users/haleyoro/Desktop/murray-neuroscience-lab/Processed CSVs/sheet_names.csv', header=None)
sheet_names = sheet_names_df.iloc[:,0].values
sheet_names # names of each annotation sheet

array(['2012_04_25_cell1', '2012_04_25_cell3', '2012_04_27_cell1',
       '2012_06_22_cell3', '2012_06_25_cell3', '2012_06_29',
       '2012_08_01_cell1', '2012_08_01_cell3', '2012_08_07',
       '2012_08_31_cell1', '2012_08_31_cell2', '2012_08_31_cell3',
       '2012_08_31_cell4', '2012_10_04_cell2', '2012_12_03_cell1',
       '2012_12_04_cell1', '2012_12_04_cell2', '2012_12_05_cell1',
       '2012_12_05_cell3', '2012_12_05_cell4', '2012_12_06_cell1',
       '2012_12_06_cell2', '2012_12_06_cell3', '2012_12_06_cell4',
       '2012_12_06_cell5', '2012_12_06_cell6', '2013_03_20_cell1',
       '2013_03_21_cell1', '2013_03_21_cell2', '2013_03_21_cell3',
       '2013_03_21_cell4', '2013_03_21_cell5', '2013_03_22_cell1',
       '2013_03_22_cell2', '2013_03_22_cell3', '2013_03_22_cell4',
       '2013_03_22_cell5', '2013_03_22_cell6', '2012_10_04_cell1'],
      dtype=object)

### All abf files available in shared folder

In [3]:
abfs = pd.read_csv('/Users/haleyoro/Desktop/murray-neuroscience-lab/Excel processor/all_abf_files.csv',header=None)
abfs = abfs[0].to_numpy()
abfs 

array(['2013_03_22_0045.abf', '2013_03_22_0042.abf',
       '2013_03_22_0039.abf', ..., '2012_08_29_0009.abf',
       '2012_08_29_0016.abf', '2012_08_29_0018.abf'],
      shape=(1050,), dtype=object)

Check to see duplicates

In [43]:
pattern = r"\d{4}_\d{2}_\d{2}_\d{4}\.abf"

abfs = [f for f in abfs if re.fullmatch(pattern, f)]

print(len(abfs)) 
print(len(set(abfs)))


1047
794


All traces in each sheet are unique

In [46]:
all_trace_names = []
for sheet in sheet_names:
    file_path = "/Users/haleyoro/Desktop/murray-neuroscience-lab/Processed CSVs/"+ sheet + '.csv'
    df_filtered = pd.read_csv(file_path)
    trace_names = df_filtered["Trace name"].unique().tolist()
    all_trace_names.extend(trace_names)

print(len(all_trace_names)==len(set(all_trace_names)))
print(len(all_trace_names))


True
158


Note: only 158 out of 794 abf files have annotations 

In [45]:
missing_abfs = pd.read_csv('/Users/haleyoro/Desktop/murray-neuroscience-lab/Excel processor/missing_files.csv',header=None)
missing_abfs = missing_abfs[0].to_numpy()
print(len(missing_abfs))


439


In [48]:
pattern = r"\d{4}_\d{2}_\d{2}_\d{4}\.abf"

missing_abfs = [f for f in missing_abfs if re.fullmatch(pattern, f)]

print(len(missing_abfs)==len(set(missing_abfs)))
print(len(missing_abfs))


True
436


['2012_10_04_0014']

Find which traces are missing in the downloaded folder:

In [39]:
missing_abfs_base = [name.replace(".abf", "") for name in missing_abfs]

matches_missing = [name for name in all_trace_names if name in missing_abfs_base]

print(f"{len(matches_missing)} matches found:")
print(matches_missing)

77 matches found:
['2012_04_25_0004', '2012_04_25_0005', '2012_04_25_0006', '2012_04_25_0000', '2012_04_25_0001', '2012_04_25_0002', '2012_04_25_0030', '2012_04_25_0031', '2012_04_25_0032', '2012_04_25_0033', '2012_04_25_0027', '2012_04_25_0028', '2012_04_25_0029', '2012_04_27_0003', '2012_04_27_0004', '2012_04_27_0005', '2012_04_27_0001', '2012_04_27_0002', '2012_04_27_0006', '2012_04_27_0007', '2012_06_22_0012', '2012_06_22_0014', '2012_06_25_0041', '2012_06_25_0042', '2012_06_29_0005', '2012_06_29_0006', '2012_06_29_0007', '2012_06_29_0001', '2012_06_29_0002', '2012_06_29_0003', '2012_08_31_0041', '2012_08_31_0042', '2012_08_31_0043', '2012_08_31_0038', '2012_08_31_0039', '2012_08_31_0037', '2012_10_04_0029', '2012_10_04_0030', '2012_10_04_0025', '2012_10_04_0026', '2012_10_04_0027', '2012_10_04_0023', '2012_10_04_0024', '2012_12_03_0002', '2012_12_04_0007', '2012_12_04_0008', '2012_12_04_0021', '2012_12_05_0013', '2012_12_05_0008', '2012_12_05_0004', '2012_12_05_0005', '2012_12_05_

Check to see that all the sheet traces have abf files

In [53]:
abfs_base = [name.replace(".abf", "") for name in abfs]

matches = [name for name in all_trace_names if name in abfs_base]

print(f"{len(matches)} matches found:")
print(matches)

157 matches found:
['2012_04_25_0004', '2012_04_25_0005', '2012_04_25_0006', '2012_04_25_0000', '2012_04_25_0001', '2012_04_25_0002', '2012_04_25_0030', '2012_04_25_0031', '2012_04_25_0032', '2012_04_25_0033', '2012_04_25_0027', '2012_04_25_0028', '2012_04_25_0029', '2012_04_27_0003', '2012_04_27_0004', '2012_04_27_0005', '2012_04_27_0001', '2012_04_27_0002', '2012_04_27_0006', '2012_04_27_0007', '2012_06_22_0012', '2012_06_22_0014', '2012_06_25_0041', '2012_06_25_0042', '2012_06_29_0005', '2012_06_29_0006', '2012_06_29_0007', '2012_06_29_0001', '2012_06_29_0002', '2012_06_29_0003', '2012_08_01_0009', '2012_08_01_0010', '2012_08_01_0011', '2012_08_01_0005', '2012_08_01_0006', '2012_08_01_0007', '2012_08_01_0008', '2012_08_01_0002', '2012_08_01_0003', '2012_08_01_0075', '2012_08_01_0076', '2012_08_01_0077', '2012_08_01_0078', '2012_08_01_0071', '2012_08_01_0072', '2012_08_01_0073', '2012_08_01_0074', '2012_08_01_0081', '2012_08_01_0082', '2012_08_01_0069', '2012_08_01_0070', '2012_08_07

In [57]:
print(all_trace_names)
print(matches)

['2012_04_25_0004', '2012_04_25_0005', '2012_04_25_0006', '2012_04_25_0000', '2012_04_25_0001', '2012_04_25_0002', '2012_04_25_0030', '2012_04_25_0031', '2012_04_25_0032', '2012_04_25_0033', '2012_04_25_0027', '2012_04_25_0028', '2012_04_25_0029', '2012_04_27_0003', '2012_04_27_0004', '2012_04_27_0005', '2012_04_27_0001', '2012_04_27_0002', '2012_04_27_0006', '2012_04_27_0007', '2012_06_22_0012', '2012_06_22_0014', '2012_06_25_0041', '2012_06_25_0042', '2012_06_29_0005', '2012_06_29_0006', '2012_06_29_0007', '2012_06_29_0001', '2012_06_29_0002', '2012_06_29_0003', '2012_08_01_0009', '2012_08_01_0010', '2012_08_01_0011', '2012_08_01_0005', '2012_08_01_0006', '2012_08_01_0007', '2012_08_01_0008', '2012_08_01_0002', '2012_08_01_0003', '2012_08_01_0075', '2012_08_01_0076', '2012_08_01_0077', '2012_08_01_0078', '2012_08_01_0071', '2012_08_01_0072', '2012_08_01_0073', '2012_08_01_0074', '2012_08_01_0081', '2012_08_01_0082', '2012_08_01_0069', '2012_08_01_0070', '2012_08_07_0019', '2012_08_07

In [58]:
# Define two lists
list1 = all_trace_names
list2 = matches

# Find extra elements in list1
extra_in_list1 = list(set(list1) - set(list2))

# Find extra elements in list2
extra_in_list2 = list(set(list2) - set(list1))

# Print the results
print("Extra elements in list1:", extra_in_list1)
print("Extra elements in list2:", extra_in_list2)


Extra elements in list1: ['2012_03_21_0027']
Extra elements in list2: []


['2012_03_21_0027'] in annotations files, but not in folder

In [59]:
trace_names_with_ext = [name + ".abf" for name in all_trace_names]

# Save to CSV
df = pd.DataFrame(trace_names_with_ext, columns=["filename"])
df.to_csv("abf_trace_names.csv", index=False)